Inject compiler dependency into model

becca-bailey opened the pull request:
View is able to access compiler through model rather than
instantiating a new HandlebarsCompiler. Currently, the MenuView
still instantiates a new compiler, as it does not have a model.

becca-bailey committed:
Inject compiler dependency into model

View is able to access compiler through model rather than
instantiating a new HandlebarsCompiler. Currently, the MenuView
still instantiates a new compiler, as it does not have a model.

becca-bailey committed:
Extract router into its own class

becca-bailey committed:
Test GameView with a MockCompiler

cdemyanovich commented on spec/GameSpec.js:
```diff
@@ -18,6 +22,14 @@ describe("Game", function() {
+  it("can be initialized with a compiler", function() {
+    expect(game.get('compiler')).toBeDefined();
+  });
+
```
Are these tests worth keeping?

becca-bailey replied:
Probably not... they were just the first tests I wrote to make sure nothing was broken.

cdemyanovich commented on spec/mocks/MockCompiler.coffee:
```diff
@@ -0,0 +1,57 @@
+  appendToContainer: (container, template) ->
+    $(container).html(template)
+
+  getHTMLForTemplateName: (templateName) ->
```
Would you have to change this markup if that of the views changes? If so, this is a form of duplication. How can you eliminate it?

becca-bailey replied:
Yeah, at the moment this is a less than ideal solution. It was a quick fix to avoid importing my partials using AJAX, because making an asynchronous AJAX request, even for files in my own directory was causing inconsistent test failures.

beccanelson commented on spec/mocks/MockCompiler.coffee:
```diff
@@ -0,0 +1,57 @@
+  appendToContainer: (container, template) ->
+    $(container).html(template)
+
+  getHTMLForTemplateName: (templateName) ->
```
It might be easier to pass in a function to apply to the template rather than specifying to use .html. In the future, I might need to use .append or something else.

becca-bailey committed:
Turn off async option for HandlebarsCompiler

This eliminates the need for a MockCompiler, and allows the tests
to run only after the partials have been loaded.
